### PR TITLE
adjustments to objdump call 

### PR DESCRIPTION
**Pull request by GitMensch**
After recognizing that the process path of objdump is resolved first, I've adjusted the PR to use that (so the additional `stat` calls seen in #527 are not needed).

* invoke directly from the processPath already resolved before
* improve error message for objdump invocation issues helping to find problems related to the call

### CODE REVIEW

**Comment by lievenhey** [comment]
@GitMensch lgtm, but can you ran `clang-format` so that the ci no longer complains?

**Comment by GitMensch** [comment]
applied `clang-format -i` and pushed

File: src/models/disassemblyoutput.cpp [hunk 1]
```diff
@@ -308,22 +308,26 @@ DisassemblyOutput DisassemblyOutput::disassemble(const QString& objdump, const Q
     }
     arguments.append(binary);
 
-    asmProcess.start(objdump, arguments);
+    asmProcess.start(processPath, arguments);
 
     if (!asmProcess.waitForStarted()) {
-        disassemblyOutput.errorMessage += QApplication::translate("DisassemblyOutput", "Process was not started.");
+        disassemblyOutput.errorMessage += QApplication::translate("DisassemblyOutput", "Process was not started.")
```
Comment (milianw) [inline src/models/disassemblyoutput.cpp:314]: string puzzle, this is impossible to translate properly. Make this one large string and then use arg to fill in the placeholders and also include the `QProcess::errorString`

```
..., "<qt>Process failed to start: <tt>%1 %2</tt> returned <tt>%2</tt>").arg(processPath, arguments.join(...), asmProcess.errorString())
```

File: src/models/disassemblyoutput.cpp [hunk 2]
```diff
@@ -308,22 +308,26 @@ DisassemblyOutput DisassemblyOutput::disassemble(const QString& objdump, const Q
...
+            + QStringLiteral("\n")
+            + QApplication::tr("command: %1 %2").arg(processPath, arguments.join(QLatin1Char(' ')));
         return disassemblyOutput;
     }
 
     if (!asmProcess.waitForFinished()) {
         disassemblyOutput.errorMessage +=
-            QApplication::translate("DisassemblyOutput", "Process was not finished. Stopped by timeout");
+            QApplication::translate("DisassemblyOutput", "Process was not finished. Stopped by timeout")
```
Comment (milianw) [inline src/models/disassemblyoutput.cpp:322]: dito

**Review by milianw — CHANGES_REQUESTED** [review]

File: src/models/disassemblyoutput.cpp (hunk 2 above)
Comment (GitMensch) [reply src/models/disassemblyoutput.cpp:322]: will fix the string puzzles, I likely was concerned about not breaking the existing translations

**Comment by GitMensch** [comment]
* string puzzle fixed, other invocations of error messages changed accordingly
* additional commit fixing a related issue (missing check -> error message)

File: src/models/disassemblyoutput.cpp [hunk 3]
```diff
@@ -272,7 +272,9 @@ DisassemblyOutput DisassemblyOutput::disassemble(const QString& objdump, const Q
     const auto processPath = QStandardPaths::findExecutable(objdump);
     if (processPath.isEmpty()) {
         disassemblyOutput.errorMessage =
-            QApplication::tr("Cannot find objdump process %1, please install the missing binutils package for arch %2")
+            QApplication::translate("DisassemblyOutput",
```
Comment (milianw) [inline src/models/disassemblyoutput.cpp:283]: these hunks here are unrelated to what the commit message talks about - split that up into a separate commit

File: src/models/disassemblyoutput.cpp [hunk 4]
```diff
@@ -303,27 +305,32 @@ DisassemblyOutput DisassemblyOutput::disassemble(const QString& objdump, const Q
...
     if (binary.isEmpty()) {
         disassemblyOutput.errorMessage +=
-            QApplication::translate("DisassemblyOutput", "Could not find %1").arg(symbol.binary);
+            QApplication::translate("DisassemblyOutput", "<qt>Could not find <tt>%1</tt>").arg(symbol.binary);
         return disassemblyOutput;
     }
     arguments.append(binary);
 
-    asmProcess.start(objdump, arguments);
+    asmProcess.start(processPath, arguments);
```
Comment (milianw) [inline src/models/disassemblyoutput.cpp:321]: ideally this here is also its own patch - whenever your commit message is something like "do A and B" then it's an indication that you should have split it up into two patches instead

File: src/models/disassemblyoutput.cpp [hunk 5]
```diff
@@ -265,7 +265,15 @@ DisassemblyOutput DisassemblyOutput::disassemble(const QString& objdump, const Q
     DisassemblyOutput disassemblyOutput;
     disassemblyOutput.symbol = symbol;
     if (symbol.symbol.isEmpty()) {
-        disassemblyOutput.errorMessage = QApplication::tr("Empty symbol ?? is selected");
+        disassemblyOutput.errorMessage =
```
Comment (milianw) [inline src/models/disassemblyoutput.cpp:268]: this doesn't really help - if you really want to wrap the `??` then just do that directly, no need for `.arg()`

**Review by milianw — CHANGES_REQUESTED** [review]

File: src/models/disassemblyoutput.cpp (hunk 5 above)
Comment (GitMensch) [reply src/models/disassemblyoutput.cpp:268]: That's what I did, just to get a compile warning about trigraphs., which was the reason to split that. Another option is to use inline hex values, bit that is likely not useful for the translation.

**Comment by GitMensch** [comment]
rebased and split from 2 to 3 separate commits (I do like the result more than the original, too); tested manually in the GUI using the files under "tests" (all but the timeout, checked this with the appImage and also works as expected).

Adjusted one test as needed and added some of the missing tests for the adjusted messages.

**Comment by milianw** [comment]
thanks!
